### PR TITLE
Feature fix for demo

### PR DESCRIPTION
**Pull request by PatrickDekker98**
added mostly examples for the demo

### CODE REVIEW

File: code/esp_code/tcp_client.c [hunk 1]
```diff
@@ -164,14 +165,15 @@ static void tcp_client_task(void *pvParameters) {
             ESP_LOGI(TAG, "recieved bytes over spi: %s", recvbuf);
             // set send buffer to something sane
             memset(sendbuf, 0x00, 129);
-            int err = send(sock, recvbuf, strlen(recvbuf), 0);
+            //int err = send(sock, recvbuf, strlen(recvbuf), 0);
```
Comment (TonyvanderKrogt) [inline code/esp_code/tcp_client.c]: Why is this one still in it?

File: code/src/esp_32.cpp [hunk 2]
```diff
@@ -12,11 +12,14 @@ namespace r2d2::communication {
         // set the first bite of the buffer to 0
         // only the first because the first byte of an external frame is its length
         uint8_t recv[256] = {0};
-        // wait till esp is ready
-        while (!hand_shake.read()){}
+        uint8_t send[256] = {frame.length, frame.id.octets[0], frame.id.octets[1], frame.type};
+                memcpy(send+4, frame.data, frame.length );
+
+// wait till esp is ready
+        while (!hand_shake.read()){hwlib::cout << 1;}
```
Comment (TonyvanderKrogt) [inline code/src/esp_32.cpp]: You can maybe do something other than cout << 1. Maybe you can just place continue in it.

**Review by TonyvanderKrogt — CHANGES_REQUESTED** [review]

File: code/esp_code/tcp_client.c (hunk 1 above)
Comment (PatrickDekker98) [reply code/esp_code/tcp_client.c]: this is outdated now

**Review by TonyvanderKrogt — APPROVED** [review]
I don't see any other problems in the demo code

**Pull request merged**